controllers/bonfire.js

```diff
@@ -48,14 +48,16 @@ exports.returnNextBonfire = function(req, res) {
             return elem;
         }
     });
-    req.user.save();
+    req.user.save(function(err) {
+        if (err) return next(err);
+    });
 
     var uncompletedBonfires = req.user.uncompletedBonfires;
 
     var displayedBonfires =  Bonfire.find({'_id': uncompletedBonfires[0]});
     displayedBonfires.exec(function(err, bonfire) {
         if (err) {
```

Comment:
Subsequent async methods must be done within the first async callback. So all the code after user.save must be within it's callback.